### DISCUSSION TITLE
Env variable not working on my create-react-app

### DISCUSSION THREAD

**Discussion by shahirsiraj** [vercel/community#2875] [Help] [answered]
### Summary

Hello everyone, I'm new to this and I'm having an issue with my env variable, which is my OpenAI api key. I have it set up in my code as process.env.REACT_APP_OPENAI_API_KEY and in vercel, I have it set up as REACT_APP_OPEN_API_KEY. I have tested the key and everything works when I have the key locally in my file but when I deploy without the key (replaced with the process.env statement as above), I get a 401 error, which is an authorization error. I am using Create-React-App if that helps. I've tried removing the REACT_APP prefix and a few other things, but I can't seem to get it to work. Would love some help! 

### Example

https://seif-project2-aurelius.vercel.app/sad

### Steps to Reproduce

Essentially, my env variable is not being detected in my Vercel deployment, details as above

**Comment by smaeda-ks** [comment, marked as answer]
@shahirsiraj did you redeploy your app after adding the env var to your project? Every time you make changes to your env var, you will need to redeploy to reflect that change
